controllers/storagecluster/reconcile.go

```diff
@@ -39,6 +39,7 @@ type resourceManager interface {
 
 type ocsCephConfig struct{}
 type ocsJobTemplates struct{}
+type ocsExtendCluster struct{}
```

Comment:
I think it would make more sense to include this as part of the `odsJobTemplates.ensureCreated()` function. Should just be able to add the new Template after the current one in the function.

Reply:
> I think it would make more sense to include this as part of the `odsJobTemplates.ensureCreated()` function. Should just be able to add the new Template after the current one in the function.

Hi @jarrpa 

Since ocsExtendCluster is a different flow from osdcleanupjob() that makes use of  ocsJobTemplates, i thought it would be better to create a new one for my use case of extend-cluster. 

could you elaborate? , I could only use one struct signature per ensureCreated() method., maybe  im missing something :)

Reply:
The extend cluster Job is still a Template of a Job. There should be no need to differentiate between different the Jobs, and indeed it would make sense to keep all future Job Templates together unless there is a really good reason to do otherwise.

Since the code between the two `ensureCreated` functions are almost identical, you could restructure the `ocsJobTemplates.ensureCreate()` function to loop through a set of defined Job Templates, of which there would presently only be two. You could then have the `newFooJob()` functions return both the Job spec and Template parameters and set those in the mutate function that way.

You would have to first modify the cleanup Job functions so that you can efficiently add in your job. This would be done in multiple commits:
1. Move cleanup Job parameters into `newCleanupJob()` and return both spec and parameters
2. Refactor `ocsJobTemplates.ensureCreate()` to loop through the defined Jobs (currently just one) in a generic way that reuses all the code you're currently duplicating
3. Add in your new Template
4. Add unit tests to validate your new Template